### PR TITLE
Updated documentation to use range notation syntax

### PR DESCRIPTION
**Pull request by genbattle**
Replaced outdated use of the `range(start, end)` function where appropriate with `start..end`, and tweaked the examples to compile and run with the latest rust. I also fixed two periphery compile issues in reference.md which were occluding whether there were any new errors created by these changes.

### CODE REVIEW

File: src/doc/reference.md [hunk 1]
```diff
@@ -2994,7 +2994,7 @@ Some examples of call expressions:
 # fn add(x: i32, y: i32) -> i32 { 0 }
 
 let x: i32 = add(1i32, 2i32);
-let pi: Option<f32> = "3.14".parse().ok();
+let pi: Option<f32> = "3.14".parse();
```
Comment (edwardw) [inline src/doc/reference.md]: `let pi: Result<f32, _> = "3.14".parse();` The API has been changed.

Comment (genbattle) [inline src/doc/reference.md]: When was it updated? [The documentation](http://doc.rust-lang.org/std/str/trait.StrExt.html#tymethod.parse) still seems to refer to it returning an Option<F>.

Comment (edwardw) [inline src/doc/reference.md]: It is #21718, just yesterday fresh out of the oven.

Comment (genbattle) [inline src/doc/reference.md]: I have no compiler to test against, the latest rust-nightly-bin package is (1d00c545e 2015-01-30 19:56:34 +0000). I can't fix this issue and verify that it works until the nightly is updated to include your change; currently I just get:
```
main.rs:12:13: 12:27 error: wrong number of type arguments: expected 1, found 2 [E0244]
main.rs:12     let pi: Option<f32, _> = "3.14".parse();
```

Comment (steveklabnik) [inline src/doc/reference.md]: You have `Option<` in that error message, not `Result<`.

Comment (genbattle) [inline src/doc/reference.md]: Why not open a separate PR to update all the doc examples affected by #21718 instead of cramming it into this one?

Comment (edwardw) [inline src/doc/reference.md]: That's, of course, a viable solution too. Fine with me.

**Comment by steveklabnik** [comment]
This needs a rebase now, but I'm interested in moving it forward.

**Comment by genbattle** [comment]
I know how to rebase my branch against master in my fork, but I haven't used github enough to figure out the best way to update my fork with mainline rust - github seems to suggest pulling the changes from mainline rust via a pull request into my fork. Is this the best way to go about it, or is there a cleaner/easier way to do this?

**Comment by genbattle** [comment]
No matter, figured it out, rebasing now.

**Comment by steveklabnik** [comment]
I have this setup:

```
$ git remote -v
origin  git@github.com:steveklabnik/rust.git (fetch)
origin  git@github.com:steveklabnik/rust.git (push)
upstream        https://github.com/rust-lang/rust.git (fetch)
upstream        https://github.com/rust-lang/rust.git (push)
```

and then

```
$ git fetch upstream
$ git rebase upstream/master # with all this entails...
$ git push origin -f mybranch
```

**Comment by genbattle** [comment]
After I did the rebase git insisted I pull and merge with my remote, but obviously in this case a push -f would have been the better option. It seems to have worked, although I don't know what it's done to the history :-/

**Comment by fujin** [comment]
Use the reflog to hard move your branch pointer back before you pull/merged, rebase again, check fast-forward, then `git push --force-with-lease`. ~0~

**Comment by genbattle** [comment]
I reset to before the pull/merge and force pushed, it seems to have corrected it?

**Comment by fujin** [comment]
@genbattle Indeed; Thank you.

**Comment by genbattle** [comment]
Steve, don't push this through yet. I just realized I don't think I ended
up fixing up newly created occurrences of range(start, end) when I did the
merge for the rebase, I'll get onto this in the morning and add another
commit to this PR.

**Comment by genbattle** [comment]
There was only one additional instance of the range function instead of range notation, so I've fixed it up, tested it, and added it to this branch.

**Comment by steveklabnik** [comment]
This looks great, @genbattle , except the original comment about `Result`. Do these tests pass? Other than that, this looks great, r=me

**Comment by genbattle** [comment]
@steveklabnik I've fixed the usage of StrExt.parse() that @edwardw originally commented on. I also fixed all similar occurrences in doc/trpl/guessing-game.md.

I did another rebase and tested against all of the examples in all of the files touched by this PR and fixed any errors that arose. It should be all good to go unless you have any more comments or recommendations for me?

**Comment by steveklabnik** [comment]
Nope, looks great! Thnak you! (not rolling up since this is a big ish change)

**Pull request merged**